sync/sync.go

```diff
@@ -140,7 +138,10 @@ func (sync *synchronizer) sayHello(to peer.ID) error {
 	)
 	msg.Sign(sync.valKeys)
 
-	sync.peerSet.UpdateStatus(to, peerset.StatusCodeConnected)
+	peer := sync.peerSet.GetPeer(to)
+	if !peer.IsKnownOrTrusty() {
+		sync.peerSet.UpdateStatus(to, peerset.StatusCodeConnected)
```

Comment:
@b00f 
why should we update status to `StatusCodeConnected` when peer is not Known or trusty?

Reply:
If there is a connection to a "known" peer and the node restarts, the status is not changed to "connected".